packages/core/mesh/teleport/src/muxing/balancer.ts

```diff
@@ -140,42 +115,80 @@ export class Balancer {
     return this._channels[index];
   }
 
-  private _getNextChunk(): ChunkEnvelope {
+  private _enqueueChunk(data: Uint8Array, trigger: Trigger, channelId: number) {
+    if (!this._channels.includes(channelId)) {
```

Comment:
I'm a bit confused about what `_channels` is meant to do. The `.includes` call does a linear O(n) search on the array. Can we merge _sendBuffers and channels into one map instead?

Reply:
_channels is used for round-robin'ing the channel chunks, which didn't seem great but didn't seem like it could have been the source of the bug.

I also wanted to use a single datastructure here but I wanted to be sure I could implement the round-robin algo properly.